funnel/models/project.py

```diff
@@ -189,30 +187,32 @@ def datelocation(self):
         If multi-day event across years
         > 30 Dec 2018–02 Jan 2019, Bangalore
 
-        ``datelocation_format`` always keeps ``date_upto`` format as ``–DD Mmm YYYY``.
-        Depending on the scenario mentioned below, format for ``date`` changes. Above examples
+        ``datelocation_format`` always keeps ``schedule_end_at`` format as ``–DD Mmm YYYY``.
+        Depending on the scenario mentioned below, format for ``schedule_start_at`` changes. Above examples
         demonstrate the same. All the possible outputs end with ``–DD Mmm YYYY, Venue``.
-        Only ``date`` format changes.
+        Only ``schedule_start_at`` format changes.
         """
-        datelocation_format = u"{date}–{date_upto} {year}"
-        if self.date == self.date_upto:
-            # if both dates are same, in case of single day project
-            strf_date = ""
-            datelocation_format = u"{date_upto} {year}"
-        elif self.date.month == self.date_upto.month:
-            # If multi-day event in same month
-            strf_date = "%d"
-        elif self.date.month != self.date_upto.month:
-            # If multi-day event across months
-            strf_date = "%d %b"
-        elif self.date.year != self.date_upto.year:
-            # if the start date and end dates are in different years,
-            strf_date = "%d %b %Y"
-        datelocation = datelocation_format.format(
-            date=self.date.strftime(strf_date),
-            date_upto=self.date_upto.strftime("%d %b"),
-            year=self.date.year)
-        return datelocation if not self.location else u', '.join([datelocation, self.location])
+        daterange = u""
+        if self.schedule_start_at and self.schedule_end_at:
+            daterange_format = u"{schedule_start_at}–{schedule_end_at} {year}"
+            if self.schedule_start_at == self.schedule_end_at:
+                # if both dates are same, in case of single day project
+                strf_date = ""
+                daterange_format = u"{schedule_end_at} {year}"
```

Comment:
Since you are rendering dates instead of datetimes, change the parameter names. I read this code expecting the actual `schedule_end_at` was being rendered here and only after scrolling down realised it was the date portion. Use `start_date` and `end_date`.